cli/src/main/java/ch/cyberduck/cli/TerminalPreferences.java

```diff
@@ -59,6 +60,7 @@ protected void setFactories() {
         this.setDefault("factory.vault.class", CryptoVault.class.getName());
         this.setDefault("factory.securerandom.class", FastSecureRandomProvider.class.getName());
         this.setDefault("factory.connectiontimeout.class", DisabledConnectionTimeout.class.getName());
+        this.setDefault("factory.s3.pathcontainerservice.class", S3PathContainerService.class.getName());
```

Comment:
Instead of setting the s3 path container service four times in each and every (executable) project, set in shared preferences location: https://github.com/iterate-ch/cyberduck/blob/ec399a0cf3bb638bd2eb1909de9eb4f8423f8fbd/core/src/main/java/ch/cyberduck/core/preferences/Preferences.java#L455

Reply:
@AliveDevil bad smell indeed. Doing so would introduce circular dependencies between `core` and `s3` module, that's why I resorted to adding 4 times.

Reply:
Configure with static string value `ch.cyberduck.core.s3.S3PathContainerService` to avoid dependency.